### PR TITLE
Fix disabling of USE_CUDNN when not found

### PR DESCRIPTION
**Pull request by orionr**
cc @goldsborough @smessmer

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -411,6 +411,8 @@ if(USE_CUDA)
     endif()
     if(CAFFE2_USE_CUDNN)
       list(APPEND Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS caffe2::cudnn)
+    else()
+      caffe2_update_option(USE_CUDNN OFF)
```
Comment (Yangqing) [inline cmake/Dependencies.cmake]: Quick question: what about CAFFE2_USE_CUDNN?

**Review by Yangqing — APPROVED** [review]
Approving to unblock - one minor question inline.

File: cmake/Dependencies.cmake (hunk 1 above)
Comment (orionr) [reply cmake/Dependencies.cmake]: CAFFE2_USE_CUDNN is updated in cmake/public/cuda.cmake, so we should be good.

**Comment by orionr** [comment]
Test failures are unrelated, so merging.

**Pull request merged**